src/components/DataTable/DataTable.js

```diff
@@ -228,15 +236,15 @@ class DataTable extends Component {
       'sortDirection',
     ]);
     const table = (
-      <Table {...rest} style={fixedHeader ? DataTable.FIXED_STYLE : null}>
+      <Table {...rest}>
         {this.renderHead()}
         {this.renderBody()}
       </Table>
     );
 
-    if (fixedHeader) {
+    if (stickyHeader) {
       return (
-        <div className="slds-grid slds-grid_vertical slds-scrollable">
+        <div className="slds-grid slds-grid_vertical slds-scrollable" onScroll={this.handleStickyHeader}>
```

Comment:
Please `debounce` this event listeners since React [does not support passive event listeners](https://github.com/facebook/react/issues/6436) yet and we should not spam scroll events. Scroll event listeners are a notorious performance bottleneck.

You can bind a debounced handler like:

``` js
constructor(props) {
  this.handleStickyHeader = debounce(this.handleStickyHeader, 300);
}
```

As for a `debounce` implementation, there is a couple of ways you can go:

* Use Lodash's `debounce` and set a short timeout like 50/100
* Use a `rAF`-based debounce (Better in theory)

I recently implemented something along those lines for a personal project, the code looks like

```js 
/**
 * Simple requestAnimationFrame-based throttle
 * @param {Function} fn callback
 */
export const throttle = (fn) => {
  let running = false;

  return () => {
    if (running) return;

    running = true;
    requestAnimationFrame(() => {
      fn();
      running = false;
    });
  };
};
```

If you go down that route, please make sure to test this properly as I am not sure that I did :stuck_out_tongue: